File: backend/src/main/java/com/tacs/backend/model/Event.java
package com.tacs.backend.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.IndexDirection;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import java.util.Date;
import java.util.Set;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document("events")
public class Event {
    @Id
    private String id;

    private String name;

    private String description;

    @Field("owner_user")
    private User ownerUser;

    private Status status;

    @Field("event_options")
    @DBRef
    private Set<EventOption> eventOptions;

    @Field("registered_users")
    @DBRef
    private Set<User> registeredUsers;

    @Field("create_date")
    @Indexed(direction = IndexDirection.DESCENDING)
    private Date createDate;


    public enum Status {
        /**
         * vote pending
         */
        VOTE_PENDING,
        /**
         * vote closed
         */
        VOTE_CLOSED,
        /**
         * pending
         */
        PENDING,
        /**
         * started
         */
        STARTED,
        /**
         * finished
         */
        FINISHED
    }
}

File: backend-go/internal/core/domain/entity/event.go
package entity

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
	"time"
)

type Event struct {
	Id              primitive.ObjectID `json:"id,omitempty" bson:"_id,omitempty"`
	Name            string             `bson:"name" binding:"required"`
	Description     string             `json:"description,omitempty" bson:"description,omitempty"`
	Status          string             `json:"status,omitempty" bson:"status,omitempty"`
	EventOptions    []EventOption      `bson:"event_options,omitempty" binding:"required"`
	OwnerUser       User               `json:"owner_user,omitempty" bson:"owner_user,omitempty" binding:"-"`
	RegisteredUsers []User             `json:"registered_users,omitempty" bson:"registered_users,omitempty" binding:"-"`
	CreateDate      time.Time          `json:"create_date,omitempty" bson:"create_date,omitempty"`
}
